main.cpp

```diff
@@ -32,28 +32,286 @@ Purpose:  This project continues developing Project3.
+    ~Wurlitzer()
+    {
+        std::cout << "Wurlitzer Destroyed\n";
+    }
+
+    void changeVolume ( int knobPosition );
+    void play ( int note = 23 );
+    void power ( bool switchPosition = true );
+    void printCondition();
+    int swellVolume(int amount, int time);
+};
+
+Wurlitzer::Wurlitzer() :
+volumeLevel ( 6 ),
+vibratoAmount ( .5 ),
+model ( "200a" ),
+condition ( "good" ),
+numKeys ( 49 ),
+output ( 23 ),
+on ( false )
+{
+
+}
+
+void Wurlitzer::changeVolume( int knobPosition ) 
+{
+    volumeLevel = knobPosition;
+    std::cout << "Knob Position is: " << knobPosition << "\n";
+}
+
+void Wurlitzer::play( int note ) 
+{
+    output = note;
+}
+
+void Wurlitzer::power( bool switchPosition ) 
+{
+    on = switchPosition;
+}
+
+void Wurlitzer::printCondition()
+{
+    std::cout << "This Wutlizer is in " << condition << " condition\n"; 
+}
 
+int Wurlitzer::swellVolume(int amount, int time)
+{
+    int swell = 0;
+    for (int i = 0; i <= time ; i += 1)
+    {
+        swell += 20;
+        if (swell >= amount)
+        {
+            std::cout << "Swell finished" << std::endl;
+            return swell;
+        }
+    }
+    return swell;
+}
 /*
  copied UDT 3:
  */
+struct ElectricBass
+{
+    int numStrings = 4;
+    std::string pickup = "precision";
+    std::string brand = "fender";
+    std::string model = "precision";
+    int yearMade = 1974;
+    int output = 23;
+    int cost = 1544;
+    int timeSinceRestrung = 13;
+    bool timeToRestring = false;
+    int tuning = 442;
+    
+    ElectricBass();
+    ~ElectricBass()
+    {
+        std::cout << "Wurlitzer Destroyed\n";
+    }
+
+    void play ( int note =  18 );
+    void sold ( int price = 1200 );
+    void restring ( bool restring = true );
+    void checkStrings (int timesPlayed);
+};
+
+ElectricBass::ElectricBass(){}
+
+void ElectricBass::play( int note ) 
+{
+    output = note;
+}
+
+void ElectricBass::sold( int price ) 
+{
+    cost = price;
+    std::cout << "This Bass was sold for $" << price << "\n";
+}
+
+void ElectricBass::restring( bool restring ) 
+{
+    if (restring)
+    {
+        timeSinceRestrung = 0;
+    }
+}
 
+void ElectricBass::checkStrings (int timesPlayed)
+{
+    for (int i = 0; i <= timesPlayed; i += 1)
+    {
+        timeSinceRestrung += 1; 
+    }
+
+    if (timeSinceRestrung > 60)
+    {
+        timeToRestring = true;
+        std::cout << "Time to change strings!" << std::endl;
+    }
+    else
+    {
+        std::cout << "Strings are still good!" << std::endl;
+    }
+} 
 /*
  new UDT 4:
  with 2 member functions
  */
+struct band
+{
+    Wurlitzer keys;
+    ElectricBass bass;
+
+    void tune()
+    {
+        keys.tuning = bass.tuning;
+    }
+
+    void turnDown(int amount)
+    {
+        keys.output -= amount;
+        bass.output -= amount;
+    }
+
+    band();
+    ~band();
+};
+
+band::band(){}
+band::~band()
+{
+        if (keys.on)
+        {
+            std::cout << "Turn off Wurlitzer\n";
+        }
+}
```

Comment:
Fix this indentation.  check the ReadMe.md for the coding style guide for this course if needed
- [ ]  revise